bundles/org.openhab.binding.pjlinkdevice/src/main/java/org/openhab/binding/pjlinkdevice/internal/PJLinkDeviceHandler.java

```diff
@@ -0,0 +1,313 @@
+                    logger.warn("Received unknown power command" + command);
+                }
+            }
+            if (channelUID.getId().equals(PJLinkDeviceBindingConstants.CHANNEL_INPUT)
+                    || channelUID.getId().equals(PJLinkDeviceBindingConstants.CHANNEL_INPUT_DYNAMIC)) {
+                if (command == RefreshType.REFRESH) {
+
+                    StringType input = new StringType(device.getInputStatus().getResult().getValue());
+                    updateState(PJLinkDeviceBindingConstants.CHANNEL_INPUT, input);
+                    updateState(PJLinkDeviceBindingConstants.CHANNEL_INPUT_DYNAMIC, input);
+
+                } else {
+                    logger.trace("Received input command" + command);
+                    Input input = new Input(((StringType) command).toString());
+                    device.setInput(input);
+                }
+            }
+
+            boolean isAudioMute = channelUID.getId().equals(PJLinkDeviceBindingConstants.CHANNEL_AUDIO_MUTE);
+            boolean isVideoMute = channelUID.getId().equals(PJLinkDeviceBindingConstants.CHANNEL_VIDEO_MUTE);
+            if (isVideoMute || isAudioMute) {
+                if (command == RefreshType.REFRESH) {
+                    // refresh both video and audio mute, as it's one request
+                    MuteQueryResponseValue muteStatus = device.getMuteStatus();
+                    updateState(PJLinkDeviceBindingConstants.CHANNEL_AUDIO_MUTE,
+                            muteStatus.isAudioMuted() ? OnOffType.ON : OnOffType.OFF);
+                    updateState(PJLinkDeviceBindingConstants.CHANNEL_VIDEO_MUTE,
+                            muteStatus.isVideoMuted() ? OnOffType.ON : OnOffType.OFF);
+                } else {
+                    if (isAudioMute) {
+                        logger.trace("Received audio mute command" + command);
+                        boolean muteOn = command == OnOffType.ON;
+                        device.setMute(MuteInstructionChannel.AUDIO, muteOn);
+                    }
+                    if (isVideoMute) {
+                        logger.trace("Received video mute command" + command);
+                        boolean muteOn = command == OnOffType.ON;
+                        device.setMute(MuteInstructionChannel.VIDEO, muteOn);
+                    }
+                }
+            }
+
+            updateStatus(ThingStatus.ONLINE);
+            logger.trace("Successfully handled command {} on channel {}", command, channelUID.getId());
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (ResponseException e) {
+            logger.error(e.getMessage());
+        } catch (AuthenticationException e) {
+            this.handleAuthenticationException(e);
+        }
+
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(PJLinkDeviceConfiguration.class);
+        setupDevice();
```

Comment:
You should consider running the setupDevice in a thread as it will establish a connection with the projector and so it could take some time.

Reply:
Could you point me to an example showing which flavour of thread creation is preferred in OpenHAB?